org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/validation/XbaseUIValidator.java

```diff
@@ -222,14 +228,38 @@ protected IClasspathEntry getResolvedClasspathEntry(IJavaProject javaProject, /*
 		IClasspathEntry result = null;
 		JavaProject castedProject = (JavaProject) javaProject;
 		castedProject.getResolvedClasspath(); // force the resolved entry cache to be populated
-		@SuppressWarnings("rawtypes")
-		Map rootPathToResolvedEntries = castedProject.getPerProjectInfo().rootPathToResolvedEntries;
+		Map<IPath, IClasspathEntry> rootPathToResolvedEntries = getRootPathToResolvedEntries(castedProject.getPerProjectInfo());
 		if (rootPathToResolvedEntries != null) {
-			result = (IClasspathEntry) rootPathToResolvedEntries.get(root.getPath());
+			result = rootPathToResolvedEntries.get(root.getPath());
 			if (result == null)
-				result = (IClasspathEntry) rootPathToResolvedEntries.get(root.getJavaProject().getPath());
+				result = rootPathToResolvedEntries.get(root.getJavaProject().getPath());
 		}
 		
 		return result;
 	}
+	
+	
+	protected final boolean isJdtCoreVersionAtLeast3390 = JavaCore.getPlugin().getBundle().getVersion().compareTo(new Version(3, 39, 0)) >= 0;
```

Comment:
Is there any reason why this isn't a private static final constant? I know it's nit picky, but I'm just wondering?

Reply:
in case somebody wants to override the method below

Reply:
That's a reason for making it protected.  Why not make it static?

Reply:
The validator is very long-living (one instance per Xbase language that is installed). I don't mind making this static or not, but it won't make much of a difference.